### PR TITLE
Fix UI texture atlas with offset

### LINKED ISSUE #11219: Texture atlases with offset don't work properly in UI
## Bevy version

0.12.1
recent main (0275508 is the last recent commit where texture atlases in UI aren't broken by a different bug)

## Relevant system information

```
AdapterInfo { name: "Apple M1 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
SystemInfo { os: "MacOS 13.6.2 ", kernel: "22.6.0", cpu: "", core_count: "10", memory: "64.0 GiB" }
```
`scale_factor` of `1.0` and `2.0` were tested.

## What you did

I first noticed this during Bevy Jam 4 as a slight offset in some of my UI images. Below you can see an example, with adjacent sprites showing at the right and bottom edges of the `E` and the right edge of the `F`. This is 103x50 texture atlas with 1px of offset and padding. This seemed to get worse at the sprites get further right/down in the sheet.

bleed

Here's an example that shows the same exact `TextureAtlas` both as a sprite and in the UI.

```rust
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest()))
        .add_systems(Startup, setup)
        .add_systems(Update, increment_atlas_index)
        .run();
}

fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut texture_atlas_layouts: ResMut<Assets<TextureAtlasLayout>>,
) {
    commands.spawn(Camera2dBundle::default());

    let texture_handle = asset_server.load("sheettest.png");
    let texture_atlas = TextureAtlasLayout::from_grid(
        UVec2::new(16, 16),
        3,
        4,
        Some(UVec2::splat(16)),
        Some(UVec2::splat(40)),
    );
    let texture_atlas_handle = texture_atlas_layouts.add(texture_atlas);

    commands.spawn(AtlasImageBundle {
        style: Style {
            width: Val::Px(256.),
            height: Val::Px(256.),
            ..default()
        },
        image: texture_handle.clone().into(),
        texture_atlas: TextureAtlas::from(texture_atlas_handle.clone()),
        ..default()
    });

    commands.spawn(SpriteSheetBundle {
        texture: texture_handle,
        atlas: TextureAtlas::from(texture_atlas_handle),
        transform: Transform::from_scale(Vec3::splat(16.0)),
        ..default()
    });
}

fn increment_atlas_index(
    mut atlases: Query<&mut TextureAtlas>,
    keyboard: Res<ButtonInput<KeyCode>>,
) {
    let frames = 12;
    if keyboard.just_pressed(KeyCode::Space) {
        for mut atlas in &mut atlases {
            atlas.index = (atlas.index + 1) % frames;
        }
    }
}
```

And `sheettest.png`. It has exaggerated offset/padding with those areas highlighted for debugging purposes.

sheettest

## What went wrong

Both images should look the same. Instead, we see this:

Screenshot 2024-01-04 at 2 05 26 PM

The image in the middle is the sprite and looks correct. 

## Additional information

This works as expected with a texture atlas without offset/padding such as the one used in the `ui_texture_atlas` example.

### PR DESCRIPTION
**Pull request by s-puig**
# Objective

- Fixes #11219 

## Solution

- Scaling calculations use texture dimensions instead of layout dimensions. 

## Testing

- Did you test these changes? If so, how?

All UI examples look fine.

- How can other people (reviewers) test your changes? Is there anything specific they need to know?

Example in #11219

## Migration Guide

```diff
let ui_node = ExtractedUiNode {
                    stack_index,
                    transform,
                    color,
                    rect,
                    image,
-                   atlas_size: Some(atlas_size * scale_factor),      
+                   atlas_scaling: Some(Vec2::splat(scale_factor)),
                    clip,
                    flip_x,
                    flip_y,
                    camera_entity,
                    border,
                    border_radius,
                    node_type,
                },
```

```diff
let computed_slices = ComputedTextureSlices {
    slices,
-    image_size,
}
```

### CODE REVIEW

File: crates/bevy_ui/src/render/mod.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ pub struct ExtractedUiNode {
     pub color: LinearRgba,
     pub rect: Rect,
     pub image: AssetId<Image>,
-    pub atlas_size: Option<Vec2>,
+    pub atlas_scaling: Option<Vec2>,
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/render/mod.rs:164]: This is the breaking change: should be simple enough to cover.

File: examples/ui/ui_texture_atlas_slice.rs [hunk 2]
```diff
@@ -50,7 +50,8 @@ fn setup(
     mut texture_atlases: ResMut<Assets<TextureAtlasLayout>>,
 ) {
     let texture_handle = asset_server.load("textures/fantasy_ui_borders/border_sheet.png");
-    let atlas_layout = TextureAtlasLayout::from_grid(UVec2::new(50, 50), 6, 6, None, None);
+    let atlas_layout =
```
Comment (KirmesBude) [inline examples/ui/ui_texture_atlas_slice.rs:53]: Looking at the image, this is probably correct, but is unrelated to the other changes.

**Review by KirmesBude — APPROVED** [review]
I ran into this in the jam and this looks good to me and works as expected in my testing.
Would be nice to have some example actually using the offset, but I do not think any existing asset has an offset.

**Review by alice-i-cecile — APPROVED** [review]

File: examples/ui/ui_texture_atlas_slice.rs (hunk 2 above)
Comment (ManevilleF) [reply examples/ui/ui_texture_atlas_slice.rs:53]: It is but the slicer `border_rect` should be increased from `22` to `24` @s-puig , the fifth texture in the sheet is drooling on the border

**Pull request merged**